modules/core/include/opencv2/core/hal/intrin_sse.hpp

```diff
@@ -1924,7 +1924,11 @@ OPENCV_HAL_IMPL_SSE_EXPAND(v_int32x4,  v_int64x2,   int,      _v128_cvtepi32_epi
 #define OPENCV_HAL_IMPL_SSE_EXPAND_Q(_Tpvec, _Tp, intrin)  \
     inline _Tpvec v_load_expand_q(const _Tp* ptr)          \
     {                                                      \
-        __m128i a = _mm_cvtsi32_si128(*(const int*)ptr);   \
+        int tmp;                                           \
+        /* Copy misaligned ptr to aligned &tmp */          \
+        /* (memcpy() should be optimized out) */           \
+        memcpy(&tmp, ptr, 4);                              \
+        __m128i a = _mm_cvtsi32_si128(tmp);                \
```

Comment:
Slowdown code execution to eliminate compiler warning is not an acceptable way.

> memcpy() should be optimized out

Please provide evidences, e.g. on https://godbolt.org/

Reply:
https://godbolt.org/z/PxPzd8h11
Are there specific compiler flags for OpenCV? I tried default GCC and Clang configurations: `PrintMisalignedPtr()` and `PrintAlignedPtr()` have the same assembly body.
ARM GCC even noticed `@ unaligned`.